starlite/datastructures/response_containers.py

```diff
@@ -213,11 +240,12 @@ def to_response(
         from starlite.exceptions import ImproperlyConfiguredException
         from starlite.response import TemplateResponse
 
+        context = self.context or {}
         if not app.template_engine:
             raise ImproperlyConfiguredException("Template engine is not configured")
         return TemplateResponse(
             background=self.background,
-            context=self.context,
+            context={**context, "request": request},
```

Comment:
you dont need to hardcode it here in a way that will overide the user. I would suffest doing this:

```python
context = self.context or {}
context.setdefault(request=request)
```

Reply:
Uhm, as a user I would prefer to be sure that some template globals won't be overridden accidentally allowing me to simplify jinja2/make function code and some of my tests.

Reply:
ok.

Reply:
@Goldziher sorry, I am a bit tired and didn't understand your last comment.

Do you want me to leave it as is (forbidding the user to override `request` global in templates) or remove overriding? If you want me to change the behavior I will do it tomorrow.

Reply:
Ok. Let's go with your preference